JavadocStyleGuide.md

```diff
@@ -0,0 +1,242 @@
+| 4 | `@see`        | Adds a “See Also” heading with a link or text entry that points to a reference | `@see string`<br>`@see <a href=”URL#value”>label</a>`<br>`@see package.class#member label` |
+| 5 | `@since`      | Adds a “Since” heading                                                         | Include the Gradle version if applicable                                                   |
+| 6 | `@deprecated` | Adds a comment indicating that this API should no longer be used               | Make sure to have an alternative API linked                                                |
+
+Custom Gradle tags:
+
+| # | Tag         | Usage                                        | Notes                                                                                                                                                                                                               |
+|---|-------------|----------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1 | `@apiSpec`  | Adds a “API Requirements” heading            | A description that applies equally to all valid implementations of the method, including preconditions, postconditions, etc.                                                                                        |
+| 2 | `@apiNote`  | Adds a “API Note” heading                    | A commentary, rationale, or example pertaining to the API.                                                                                                                                                          |
+| 3 | `@implSpec` | Adds a “Implementation Requirements” heading | This is where the default implementation (or an overrideable implementation in a class) is specified.                                                                                                               |
+| 4 | `@implNote` | Adds a “Implementation Note” heading         | This section contains informative notes about the implementation, such as advice to implementors, or performance characteristics that are specific to the implementation in this class of this version of the JDK.  |
+
+When a block tag doesn't fit on a single line, continuation lines are indented four (or more) spaces from the position of the `@`.
+
+## 1.2 The summary fragment
+
+Each Javadoc block begins with a brief summary fragment - this is the first sentence up until the character `.` is encountered.
+
+This fragment is very important: it is the only part of the text that appears in certain contexts such as class and method indexes:
+
+```java
+/**
+ * Returns an Image object that can then be painted on the screen.
+ */
+```
+
+## 1.3 Code blocks and snippets
+
+`<pre>` is the default HTML tag for preformatted text.
```

Comment:
I think our convention for multi-line snippets is actually `<pre>{@code ...... }</pre>` and that's what it's also recommended: https://reflectoring.io/howto-format-code-snippets-in-javadoc/

Reply:
Yes, that is the standard convention, but I added syntax highlighting for code snippets in the Javadocs, which requires this "custom" implementation.